daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/parsers/BinaryNumberParsers.scala

```diff
@@ -168,14 +168,15 @@ abstract class BinaryIntegerBaseParser(override val context: ElementRuntimeData,
         else { dis.getUnsignedLong(nBits, start).toLong }
       }
 
-    if (!primNumeric.isValidRange(num)) {
-      PE(start, "Parsed %s is out of range for type: %s",
-        context.optPrimType.get.globalQName, num)
-      return
+    val res = try {
+      primNumeric.fromNumber(num)
+    } catch {
+      case e: Throwable => {
+        PE(start, e.getMessage)
```

Comment:
The second arg of PE is a format string. So if we getMessage ever returns anything with a %s or something in it, then this is going to fail because itw ill expect more var args. This should instead be:
```suggestion
        PE(start, "%s", e.getMessage)
```